### PR TITLE
Implement back forward buttons in the top bar - closes #1807

### LINKED ISSUE #1807: Implement Back/Forward buttons in the top bar
The buttons should work as browser history back/forward to navigate through the pages. As designed in https://netguru.invisionapp.com/share/VFQU7MABX4U#/screens

### PR DESCRIPTION
**Pull request by osvaldovega**
### What issue have I solved?

-- #1807

### How have I implemented/fixed it?
This PR include the functionality of use back and forward buttons for navigation in the app. This buttons are located in the topBar component and only are shown in the extra large and large views.

### How has this been tested?
For this the user should be login into the application and only in here the buttons will be enable in the login or registration there is no functionality.

If the user is logout the navigation will work only for those available pages.

### Review checklist
- The PR follows our [Test guide](/LiskHQ/lisk-hub/blob/development/docs/TEST_GUIDE.md)
- The PR follows our [CSS guide](/LiskHQ/lisk-hub/blob/development/docs/CSS_GUIDE.md)

### CODE REVIEW

File: src/components/topBar/navigationButtons.css [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+      cursor: default;
+    }
+  }
+
+  & > button:last-child {
+    margin-left: 25px;
+  }
+}
+
+@media (--medium-viewport) {
```
Comment (massao) [inline src/components/topBar/navigationButtons.css]: We should really hide it in small screens??

File: src/components/topBar/navigationButtons.js [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+    };
+
+    this.onGoBack = this.onGoBack.bind(this);
+    this.onGoForward = this.onGoForward.bind(this);
+    this.resetNavigationValues = this.resetNavigationValues.bind(this);
+  }
+
+  componentDidMount() {
+    this.setState({
+      ...this.state,
```
Comment (massao) [inline src/components/topBar/navigationButtons.js]: Don't need to destructure here.

File: src/components/topBar/navigationButtons.js [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+      firstPageIndex: this.props.history.length,
+      historyLength: this.props.history.length,
+      currentPageIndex: this.props.history.length,
+    });
+  }
+
+  resetNavigationValues() {
+    this.setState({
+      ...this.state,
```
Comment (massao) [inline src/components/topBar/navigationButtons.js]: same here

File: src/components/topBar/navigationButtons.js [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+  // eslint-disable-next-line max-statements
+  shouldComponentUpdate(nextProps, nextState) {
+    if (!!nextProps.account.afterLogout !== nextState.userLogout) {
+      this.resetNavigationValues();
+      return false;
+    }
+
+    if (nextProps.history.length !== nextState.historyLength && nextState.action === '') {
+      this.setState({
+        ...nextState,
```
Comment (massao) [inline src/components/topBar/navigationButtons.js]: same here

File: src/components/topBar/navigationButtons.js [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+        historyLength: nextProps.history.length,
+        currentPageIndex: nextProps.history.length,
+        action: '',
+      });
+      return false;
+    }
+
+    if (nextState.action === 'back' && nextState.currentPageIndex > nextState.firstPageIndex) {
+      this.setState({
+        ...nextState,
```
Comment (massao) [inline src/components/topBar/navigationButtons.js]: same here

File: src/components/topBar/navigationButtons.js [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+        currentPageIndex: nextState.currentPageIndex - 1,
+        action: '',
+      });
+      this.props.history.goBack();
+      return false;
+    }
+
+    if (nextState.action === 'forward' && nextState.currentPageIndex < nextProps.history.length) {
+      this.setState({
+        ...nextState,
```
Comment (massao) [inline src/components/topBar/navigationButtons.js]: same here

File: src/components/topBar/navigationButtons.js [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+    }
+
+    return true;
+  }
+
+  onGoBack(e) {
+    e.preventDefault();
+
+    this.setState({
+      ...this.state,
```
Comment (massao) [inline src/components/topBar/navigationButtons.js]: same here

File: src/components/topBar/navigationButtons.js [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+      action: 'back',
+    });
+  }
+
+  onGoForward(e) {
+    e.preventDefault();
+
+    this.setState({
+      ...this.state,
```
Comment (massao) [inline src/components/topBar/navigationButtons.js]: same here

File: src/components/topBar/navigationButtons.css (hunk 1 above)
Comment (osvaldovega) [reply src/components/topBar/navigationButtons.css]: yes based on the mockups we don't need it in the small views

File: src/components/topBar/navigationButtons.js (hunk 2 above)
Comment (osvaldovega) [reply src/components/topBar/navigationButtons.js]: done

File: src/components/topBar/navigationButtons.js (hunk 3 above)
Comment (osvaldovega) [reply src/components/topBar/navigationButtons.js]: done

File: src/components/topBar/navigationButtons.js (hunk 4 above)
Comment (osvaldovega) [reply src/components/topBar/navigationButtons.js]: done

File: src/components/topBar/navigationButtons.js (hunk 5 above)
Comment (osvaldovega) [reply src/components/topBar/navigationButtons.js]: done

File: src/components/topBar/navigationButtons.js (hunk 6 above)
Comment (osvaldovega) [reply src/components/topBar/navigationButtons.js]: done

File: src/components/topBar/navigationButtons.js (hunk 7 above)
Comment (osvaldovega) [reply src/components/topBar/navigationButtons.js]: done

File: src/components/topBar/navigationButtons.js (hunk 8 above)
Comment (osvaldovega) [reply src/components/topBar/navigationButtons.js]: done

**Pull request merged**